Modules/socketmodule.c

```diff
@@ -5574,6 +5574,23 @@ static PyType_Spec sock_spec = {
     .slots = sock_slots,
 };
 
+static PyObject *
+socket_getattr(PyObject *self, PyObject *name)
+{
+    const char *attrname = PyUnicode_AsUTF8(name);
```

Comment:
This can fail so you need to check if `attrname` is not NULL. In addition, I'd move the call before `strcmp` since you don't use it until then.